Twitter Util support

raulraja opened the pull request:
Fixes #114 

Changed scope of this PR to support Twitter Util types as supported by catbird and providing the extra Capture instances until `ApplicativeEval / MonadDefer` make it to cats and then is supported in catbird

peterneyens commented on freestyle-twitter-future/src/main/scala/future.scala:
```diff
@@ -0,0 +1,37 @@
+import cats.MonadError
+
+class FreestyleTwitterMonad extends MonadError[Future, Throwable] with Capture[Future] {
+
```
`catbird` (a project with cats type class instances for Twitter data types) does override `ap` as you did before (see [catsbird instances `Future`](https://github.com/travisbrown/catbird/blob/master/util/src/main/scala/io/catbird/util/future.scala)).

raulraja replied:
I removed it because 'ap' was never called. I'm out of a computer today but if you think it makes sense to add it feel free to commit on this branch and we'll get it merged

peterneyens commented:
If (or when) the `ApplicativeEval` / `MonadDefer` Cats PR somehow comes through and these instances are added to `catsbird`, we may need to recheck if this `freestyle-twitter-future` module brings something extra.
We could then maybe add a section to the documentation about how you can use Twitter `Future` with freestyle with the examples from the current tests.

peterneyens commented:
I should take a look later to test the laws for all the `MonadError` instances we provide.

raulraja commented:
@peterneyens makes sense. I did not know about catsbird. If those PRs make it to cats we can get rid of this module and describe how to achieve the same functionality in the docs through catsbird

raulraja commented:
@peterneyens I updated this PR to support all the datatypes supported by catbird in twitter util and added the additional Capture instances. Can you take another look when you have a chance?

peterneyens commented:
I am not sure we should provide a `Capture[Var]` instance. `Var` looks similar to `MVar` from scalaz and Monix  and I don't think that is something you would use as a target for a computation.

It also seems that it is not possible to create a stack safe `tailRecM` for `Var` which could be a problem with `Free`.

For the rest :+1:
